Web and Mac support

saifr383 opened the pull request:
In this latest update, I've included support for macOS and web platforms within the Flutter SDK.

malik19995 commented:
Hello, I've been testing this PR for macOS. 

I've configured an example project. Linked the PR version of the SDK. Facing the following issues on macOS: 

I've added the BundleURL Types entry in Info.plist 

<img width="648" alt="Screenshot 2024-03-27 at 12 10 10 AM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/22349b4c-460d-44bd-9780-691eca3d623b">

The SDK initializes without an error. But when I trigger sdk.login(), it returns the error below. 


<img width="1728" alt="Screenshot 2024-03-27 at 12 09 21 AM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/41b3f7e3-209d-4bb5-bf1f-c3207dadc878">

Please let me know if I missed something.  @saifr383

malik19995 commented:
Thank you! 

Added the entitlement. 
Still getting this error. PFA. 

<img width="1728" alt="Screenshot 2024-03-27 at 12 30 14 AM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/2319e9f3-c231-441e-8a01-a4960a7b3a09">

saifr383 commented:
This is mostly due to wrong client id and all


On Wed, 27 Mar 2024 at 12:31 AM, Hamza A.Malik ***@***.***>
wrote:

> Thank you!
>
> Added the entitlement.
> Still getting this error. PFA.
> Screenshot.2024-03-27.at.12.30.14.AM.png (view on web)
> <https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/2319e9f3-c231-441e-8a01-a4960a7b3a09>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kinde-oss/kinde-flutter-sdk/pull/14#issuecomment-2021305754>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVWU6FSDBRZTQEHMCEKQGS3Y2HEIHAVCNFSM6AAAAABE3L3NEWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMRRGMYDKNZVGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

saifr383 commented:
Is this working with starter kit?

On Wed, 27 Mar 2024 at 12:50 AM, saif rehman ***@***.***> wrote:

> This is mostly due to wrong client id and all
>
>
> On Wed, 27 Mar 2024 at 12:31 AM, Hamza A.Malik ***@***.***>
> wrote:
>
>> Thank you!
>>
>> Added the entitlement.
>> Still getting this error. PFA.
>> Screenshot.2024-03-27.at.12.30.14.AM.png (view on web)
>> <https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/2319e9f3-c231-441e-8a01-a4960a7b3a09>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kinde-oss/kinde-flutter-sdk/pull/14#issuecomment-2021305754>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AVWU6FSDBRZTQEHMCEKQGS3Y2HEIHAVCNFSM6AAAAABE3L3NEWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAMRRGMYDKNZVGQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

saifr383 commented:
> Thank you!
> 
> Added the entitlement. Still getting this error. PFA.
> 
> <img alt="Screenshot 2024-03-27 at 12 30 14 AM" width="1728" src="https://private-user-images.githubusercontent.com/46687299/317012923-2319e9f3-c231-441e-8a01-a4960a7b3a09.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE0ODMwNTIsIm5iZiI6MTcxMTQ4Mjc1MiwicGF0aCI6Ii80NjY4NzI5OS8zMTcwMTI5MjMtMjMxOWU5ZjMtYzIzMS00NDFlLThhMDEtYTQ5NjBhN2IzYTA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI2VDE5NTIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2MDM1MmY4NjQ4MzI1NmQ1NGFiMWUzNmQwZDliYmE5Y2IyOGYxNjQzMGU5YmQzNDY2ZDA0MTQwY2JkYTYzYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h9BBXViz84_ur0lfJSmHJAt12kEGEEExBr8xQ0sMMOI">

You are testing it wrong this Pr is only for web support You can test Mac Os support in other
 Pr

malik19995 commented:
Using starter-kit, I face this issue:

No logs are produced.  The app redirects to browser, but never opens anything.

I added the network permission in debug and release entitlements. 

The client ID is as provided by the Kinde portal. The complete env file is copied from there. 

Can you please tell what do you send for loginRedirectUri and logoutRedirectUri in macOS? 

https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/fd444de2-65a8-493c-aaf0-3e0ee5da67fd

malik19995 commented:
The issue was with the callback URLs. 

Fixing them, fixed the issue. Thank you for your help 🙌🏼

malik19995 commented:
'flutter test' on the SDK fails. Please add MockBuildContext to the tests.  PFA. 

<img width="1728" alt="Screenshot 2024-03-28 at 8 01 02 PM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/02a6efeb-4a63-40e9-b5c7-416bb8cafc71">

<img width="1332" alt="Screenshot 2024-03-28 at 8 12 18 PM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/3f37428a-82cc-4e45-b337-cdb6f47a2201">
 
<img width="581" alt="Screenshot 2024-03-28 at 8 12 37 PM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/741d3ccc-bbad-464a-8b8d-0d3256ab9051">

Another issue with the logout test.

saifr383 commented:
> Hello, I've been testing this PR for macOS.
> 
> I've configured an example project. Linked the PR version of the SDK. Facing the following issues on macOS:
> 
> I've added the BundleURL Types entry in Info.plist
> 
> <img alt="Screenshot 2024-03-27 at 12 10 10 AM" width="648" src="https://private-user-images.githubusercontent.com/46687299/317007226-22349b4c-460d-44bd-9780-691eca3d623b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyMjk1NjYsIm5iZiI6MTcxMjIyOTI2NiwicGF0aCI6Ii80NjY4NzI5OS8zMTcwMDcyMjYtMjIzNDliNGMtNDYwZC00NGJkLTk3ODAtNjkxZWNhM2Q2MjNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDExMTQyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjN2FkZjJjOTJkNTVmYjZhYjgxYjRmMmIzNTQ0MWZjNWFmZDQxYTliZTA4ZmZhZGJkNzNkNWU5OThiZjQ0NjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sUdSzeDr-YoWmKnj1FYmNUUD_X1URThrX85NObQVSqg">
> The SDK initializes without an error. But when I trigger sdk.login(), it returns the error below.
> 
> <img alt="Screenshot 2024-03-27 at 12 09 21 AM" width="1728" src="https://private-user-images.githubusercontent.com/46687299/317006977-41b3f7e3-209d-4bb5-bf1f-c3207dadc878.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyMjk1NjYsIm5iZiI6MTcxMjIyOTI2NiwicGF0aCI6Ii80NjY4NzI5OS8zMTcwMDY5NzctNDFiM2Y3ZTMtMjA5ZC00YmI1LWJmMWYtYzMyMDdkYWRjODc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDExMTQyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0ZWUxOTU0N2U3YzgxNTJkODVlOTZjNjIyNGRkODIyNzBiOWEyOTlmMWQ0NGI3ZThjMzYxZjM3MDc0ODIyMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nru_SuoM-ZTVWfJ0LeQCv4XSLSAv9Qr4nybu5nVvRMg">
> Please let me know if I missed something. @saifr383

Updated the code you can test it now

malik19995 commented:
Unable to Create an Organization on the Web, 

PFA error logs. 
<img width="957" alt="Screenshot 2024-04-19 at 2 54 01 PM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/9b8ac921-9a8f-44df-8a92-05a1bbe2f05a">

malik19995 commented:
https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/d907f1fd-da85-4674-af5d-10d1f4875cb3

Facing this issue on web. 

https://kinde-22dbd.web.app/ 

After login, the context does not switch properly. on starterkit

saifr383 commented:
> Screen.Recording.2024-04-21.at.8.18.33.PM.mov 
> Facing this issue on web.
> 
> https://kinde-22dbd.web.app/
> 
> After login, the context does not switch properly. on starterkit

Can you tell the whole steps to replicate it I am unable to replicate it on my end

saifr383 commented:
> Unable to Create an Organization on the Web,
> 
> PFA error logs. <img alt="Screenshot 2024-04-19 at 2 54 01 PM" width="957" src="https://private-user-images.githubusercontent.com/46687299/323923958-9b8ac921-9a8f-44df-8a92-05a1bbe2f05a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0MDUwNTUsIm5iZiI6MTcxNDQwNDc1NSwicGF0aCI6Ii80NjY4NzI5OS8zMjM5MjM5NTgtOWI4YWM5MjEtOWE4Zi00NGRmLThhOTItMDVhMWJiZTJmMDVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDE1MzIzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyNjA2YmM1NmRiZmQyZjMyNmFiOTlmOTU1YmQ0ODBjM2VkMzIxNzg5MjAyZjg3ZWM5NjA1OGNhZDYwNGYyZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.94QQXTVbG9JKAOJ_TSRjtDM_WWm48qvuEpGcX4TIwQ0">

error was bcz of context parameter empty has added a exception message for that

malik19995 commented:
Ok perfect. I'll check this out. 

I am also facing another issue on macOS. 

The callback isUserAuthenticated() does not work for macOS. Keeps returning false, even after login is completed and auth token has been obtained.

saifr383 commented:
> Ok perfect. I'll check this out.
> 
> I am also facing another issue on macOS.
> 
> The callback isUserAuthenticated() does not work for macOS. Keeps returning false, even after login is completed and auth token has been obtained.

You have to add 
<key>keychain-access-groups</key>
<array/>

in entitlements as we are using key chain for storing the token

malik19995 commented:
Perfect, works! Thank you! 

Can we please add these setup steps to the README as well?

malik19995 commented:
One more thing,  

You cannot add the `oauth_webauth` package like this. This cannot be merged with the main. Since it needs to be published to pub.dev. PFA.

I suggest either publishing the `oauth_webauth` package and linking via pub.dev or manually adding the package to the repo. 

But it should be added that the local package implementation is linked to the repo so we can pull updates. 

<img width="1728" alt="Screenshot 2024-05-03 at 4 54 50 PM" src="https://github.com/kinde-oss/kinde-flutter-sdk/assets/46687299/49baeb70-8a04-43bd-9249-790c77dafbc3">

saifr383 commented:
Published the package to Pub Dev and removed the GitHub repository link.

malik19995 commented:
I am unable to run this. I am testing with PR #4 on flutter-starter-kit. 

Please check the screenshot for the error log and Flutter doctor details. 
I believe the ViewConfiguration issue is originating from the oauth_web_plugin. I am on Flutter 3.22.2.

<img width="1137" alt="Screenshot 2024-07-12 at 10 54 20 AM" src="https://github.com/user-attachments/assets/58e84c72-9ecf-4401-89fe-9da35df184f6">

saifr383 commented:
> I am unable to run this. I am testing with PR #4 on flutter-starter-kit.
> 
> Please check the screenshot for the error log and Flutter doctor details. I believe the ViewConfiguration issue is originating from the oauth_web_plugin. I am on Flutter 3.22.2.
> 
> <img alt="Screenshot 2024-07-12 at 10 54 20 AM" width="1137" src="https://private-user-images.githubusercontent.com/46687299/348141027-58e84c72-9ecf-4401-89fe-9da35df184f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA3ODk1NzAsIm5iZiI6MTcyMDc4OTI3MCwicGF0aCI6Ii80NjY4NzI5OS8zNDgxNDEwMjctNThlODRjNzItOWVjZi00NDAxLTg5ZmUtOWRhMzVkZjE4NGY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEyVDEzMDExMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YTFhOTU3ZjYwZGNmYTQ2ZmE1YjBiOTU5ZGQ1ODc0Mjc1MjBlYzI1NGU5YzdjNjFkNWM0NjQwYWFiYmE0YzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LeaPsLPr-0HxXEqpBMdmZu3ExVfK0eWrkVuw4Rrqtmw">

Fixed this issue you can check it by updating the sdk package

malik19995 commented on lib/src/kinde_flutter_sdk.dart:
```diff
@@ -140,51 +148,137 @@ class KindeFlutterSDK with TokenUtils, HandleNetworkMixin {
-    }
-    if (_config?.audience != null) {
-      params.putIfAbsent(_audienceParamName, () => _config!.audience!);
+    if (kIsWeb && context == null) {
+      throw KindeError("context is required for web");
     }
 
-    if (type == AuthFlowType.pkce) {
-      return _pkceLogin(params);
-    } else {
-      return _normalLogin(params);
+    {
+      final params = HashMap<String, String>.from(additionalParams);
+      if (orgCode != null) {
+        params.putIfAbsent(_orgCodeParamName, () => orgCode);
+      }
+      if (_config?.audience != null) {
+        params.putIfAbsent(_audienceParamName, () => _config!.audience!);
+      }
+      if (kIsWeb) {
+        _webLogin(
+          params,
+          context!,
+        );
+      } else {
+        if (type == AuthFlowType.pkce) {
+          return _pkceLogin(params);
+        } else {
+          return _normalLogin(params);
+        }
+      }
```
@saifr383 Please adopt this.

malik19995 commented:
Perfect, this works now! Thank you! @saifr383 

I just added a comment on a code rabbit's review suggestions; please check those out and refactor them.

saifr383 commented:
@malik19995  refactored the code according to code rabbit you can  reveiw it

malik19995 commented:
Cleared for merge 💯

AndroidJedi commented on lib/src/api/o_auth_api.dart:
```diff
@@ -99,8 +99,22 @@ class OAuthApi {
+    );
+
+    return response.statusCode == 200;
+  }
```
Web/MacOs test results:

I'm able to signIn and signOut on Web and just SignIn on MacOs.
For signOut It just calls BE API /signout and on Web without any further redirection it just reloads the page:
if (kIsWeb) {
html.window.location.reload();
}

But the same code doesn't work for MacOS, as the compilers are different and for MacOs compiler this is an invalid string interpolation:
const path = r'$url';

but, again for web it works well;

AndroidJedi commented:
test result:
kinde-flutter-sdk works well on Android/iOS/WEB.

on MacOs logout doesn't work, gives error:

```
flutter: ------ OAuthWebAuth appBaseUri: file:///Users/serhiiholovashchuk/Library/Containers/au.kinde.flutterstarterkit.flutterStarterKit1/Data ------
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: DioException [unknown]: null
Error: Invalid argument(s): No host specified in URI $url
#0 DioMixin.fetch (package:dio/src/dio_mixin.dart:519:7)
<asynchronous suspension>
#1 OAuthApi.logout (package:kinde_flutter_sdk/src/api/o_auth_api.dart:108:22)
<asynchronous suspension>
#2 KindeFlutterSDK.handleWebAndMacLogout (package:kinde_flutter_sdk/src/kinde_flutter_sdk.dart:188:5)
<asynchronous suspension>
#3 KindeFlutterSDK.logout (package:kinde_flutter_sdk/src/kinde_flutter_sdk.dart:167:7)
<asynchronous suspension>
#4 *HomePageState.*signOu
```

malik19995 commented:
The issue on macOS stems from 

`https://pub.dev/packages/flutter_custom_tabs `

This plugin launches the URLs internally within the application via inAppWebView. It is not yet compatible with macOS, so there is no implementation error. 

**We can add a PR to that plugin and request them to merge and release. But this might take some time.** 

I have found an alternative setup that works for us right now. However, the only issue might be an extra redirect to the user's web browser instead. 

We need to add the following case to our SDK logout method. 

PubSpec Update:  (Add Dependency)
`  url_launcher: ^6.3.0
` 

`
else if (Platform.isMacOS) {
      // TODO: REFACTOR:: Upgraded CustomTabs with MacOS Support Plugin is required, to do the callback internally. Right now, it launches in a separate browser window.
      await launchUrl(
        Uri.parse(
          _buildEndSessionUrl().toString(),
        ),
      );
    }
    `
    This uses the default UrlLauncher from the Flutter team. 
    **This opens the URL externally and performs a successful logout.** 
    
<img width="1256" alt="Screenshot 2024-08-12 at 6 12 02 AM" src="https://github.com/user-attachments/assets/55ce70f6-cad5-4e17-81ca-afdc9446cb33">

saifr383 commented:
> The issue on macOS stems from
> 
> `https://pub.dev/packages/flutter_custom_tabs `
> 
> This plugin launches the URLs internally within the application via inAppWebView. It is not yet compatible with macOS, so there is no implementation error.
> 
> **We can add a PR to that plugin and request them to merge and release. But this might take some time.**
> 
> I have found an alternative setup that works for us right now. However, the only issue might be an extra redirect to the user's web browser instead.
> 
> We need to add the following case to our SDK logout method.
> 
> PubSpec Update: (Add Dependency) ` url_launcher: ^6.3.0`
> 
> `else if (Platform.isMacOS) { // TODO: REFACTOR:: Upgraded CustomTabs with MacOS Support Plugin is required, to do the callback internally. Right now, it launches in a separate browser window. await launchUrl( Uri.parse( _buildEndSessionUrl().toString(), ), ); }` This uses the default UrlLauncher from the Flutter team. **This opens the URL externally and performs a successful logout.**
> 
> <img alt="Screenshot 2024-08-12 at 6 12 02 AM" width="1256" src="https://private-user-images.githubusercontent.com/46687299/356917165-55ce70f6-cad5-4e17-81ca-afdc9446cb33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0NTI3MDEsIm5iZiI6MTcyMzQ1MjQwMSwicGF0aCI6Ii80NjY4NzI5OS8zNTY5MTcxNjUtNTVjZTcwZjYtY2FkNS00ZTE3LTgxY2EtYWZkYzk0NDZjYjMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEyVDA4NDY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3NDZlYWJhYjI5ZjcyMGFjNTk0YzQ1Yjg0ZjMwZWYwNjliOTNlOGY3NjQ1ZDZlYjYxOTliY2NkZjkyMGNhOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RlVCsJJIuOfo7cG317UNka42VGCwooReMvWgdka5XTY">

We can make a direct GET request instead of opening the same URL in a browser.

onderay commented:
Closing this PR in favour of the work on this PR - https://github.com/kinde-oss/kinde-flutter-sdk/pull/21
